Duplicati/UnitTest/TestUtils.cs

```diff
@@ -169,129 +170,88 @@ private static int IndexOf(List<string> lst, string m)
-                        Console.WriteLine("File differs: " + relpath);
-                    }
-                }
-            }
-
-            foreach (string s in files2)
-            {
-                Log.WriteErrorMessage(LOGTAG, "ExtraFile", null, "Extra file: {0}", s.Substring(f2.Length));
-                Console.WriteLine("Extra file: " + s.Substring(f2.Length));
-            }
-
-            if (anymissing)
-                throw new Exception("Verify failed, some items are missing");
         }
 
         /// <summary>
-        /// Compares two files by reading all bytes, and comparing one by one
+        /// Asserts that two files are equal by comparing their length, contents, and, optionally, their metadata.
         /// </summary>
-        /// <param name="f1">One file</param>
-        /// <param name="f2">Another file</param>
-        /// <param name="display">File display name</param>
-        /// <returns>True if they are equal, false otherwise</returns>
-        public static bool CompareFiles(string f1, string f2, string display, bool verifymetadata)
+        /// <param name="expectedFile">The expected file.</param>
+        /// <param name="actualFile">The actual file.</param>
+        /// <param name="verifymetadata">True to also compare file metadata.</param>
+        /// <param name="contextMessage">Context information to include in an assert message.</param>
+        public static void AssertFilesAreEqual(string expectedFile, string actualFile, bool verifymetadata, string contextMessage)
         {
-            using (System.IO.FileStream fs1 = System.IO.File.OpenRead(f1))
-            using (System.IO.FileStream fs2 = System.IO.File.OpenRead(f2))
-                if (fs1.Length != fs2.Length)
-                {
-                    Log.WriteErrorMessage(LOGTAG, "LengthsDiffer", null, "Lengths differ: {0}, {1} vs {2}", display, fs1.Length.ToString(), fs2.Length.ToString());
-                    Console.WriteLine("Lengths differ: " + display + ", " + fs1.Length.ToString() + " vs. " + fs2.Length.ToString());
-                    return false;
-                }
-                else
+            using (var expectedFileStream = SystemIO.IO_OS.FileOpenRead(expectedFile))
+            using (var actualFileStream = SystemIO.IO_OS.FileOpenRead(actualFile))
+            {
+                // Compare file lengths
+                var expectedFileStreamLength = expectedFileStream.Length;
+                var actualFileStreamLength = actualFileStream.Length;
+                Assert.That(actualFileStreamLength, Is.EqualTo(expectedFileStreamLength), $"{contextMessage}, file size mismatch for {expectedFile} and {actualFile}");
+                // Compare file contents
+                // The byte-by-byte compare is dog-slow, so we use a fast(-er) check, and then report the first byte diff if required
+                if (!Utility.CompareStreams(expectedFileStream, actualFileStream, true))
                 {

```

Comment:
I tried testing this with two files of the same length containing different bytes, and this method didn't raise a failure.  After the call to `Utility.CompareStreams`, do we need to rewind the streams by setting their `Positions` to `0`?

Reply:
Yes, I missed that.  I've added a commit for this.